### ISSUE TITLE
SSLError: EOF occurred in violation of protocol

### ISSUE THREAD

**Issue by harishpendela** [google/google-api-python-client#97]
I am using google-api-python-client_v1.2 on python 2.7.3. Often I see the following error while trying to build a drive service object. This is the stack trace:

drive_service = build('drive', 'v2', http=http)
File "/local/lib/python2.7/site-packages/google_api_python_client-1.2-py2.7.egg/oauth2client/util.py", line 132, in positional_wrapper
   return wrapped(_args, *_kwargs)
 File "/local/lib/python2.7/site-packages/google_api_python_client-1.2-py2.7.egg/apiclient/discovery.py", line 192, in build
resp, content = http.request(requested_url)
 File "/local/lib/python2.7/site-packages/google_api_python_client-1.2-py2.7.egg/oauth2client/util.py", line 132, in positional_wrapper
return wrapped(_args, *_kwargs)
 File "/local/lib/python2.7/site-packages/google_api_python_client-1.2-py2.7.egg/oauth2client/client.py", line 475, in new_request
self._refresh(request_orig)
 File "/local/lib/python2.7/site-packages/google_api_python_client-1.2-py2.7.egg/oauth2client/client.py", line 653, in _refresh
   self._do_refresh_request(http_request)
 File "/local/lib/python2.7/site-packages/google_api_python_client-1.2-py2.7.egg/oauth2client/client.py", line 682, in _do_refresh_request
   self.token_uri, method='POST', body=body, headers=headers)
File "/local/lib/python2.7/site-packages/httplib2/__init__.py", line 1593, in request
(response, content) = self._request(conn, authority, uri, request_uri, method, body, headers, redirections, cachekey)
File "/local/lib/python2.7/site-packages/httplib2/__init__.py", line 1335, in _request
(response, content) = self._conn_request(conn, request_uri, method, body, headers)
 File "/local/lib/python2.7/site-packages/httplib2/__init__.py", line 1257, in _conn_request
   self.disable_ssl_certificate_validation, self.ca_certs)
File "/local/lib/python2.7/site-packages/httplib2/__init__.py", line 80, in _ssl_wrap_socket
cert_reqs=cert_reqs, ca_certs=ca_certs)
 File "/usr/lib/python2.7/ssl.py", line 381, in wrap_socket
   ciphers=ciphers)
File "/usr/lib/python2.7/ssl.py", line 143, in __init__
   self._sslobj.do_handshake()
SSLError: [Errno 8] _ssl.c:504: EOF occurred in violation of protocol

**Comment by nathanielmanistaatgoogle** [comment]
What's got you stuck using the two-years-old version 1.2 rather than the current 1.4.0? Are you able to reproduce the problem using 1.4.0?

**Comment by harishpendela** [comment]
I recently updated my google-api-python-client to v1.4.0. I still see this issue.

**Comment by dhermes** [comment]
@nathanielmanistaatgoogle See https://github.com/GoogleCloudPlatform/gcloud-python/issues/1214 to understand what's the likely cause.

TL;DR: `httplib2.Http` is not thread-safe ([ref][1])

[1]: https://developers.google.com/api-client-library/python/guide/thread_safety

I recommend closing this out

**Comment by nathanielmanistaatgoogle** [comment]
@harishpendela, do you agree with @dhermes both about his suspected cause and that this issue should be closed?

**Comment by bantini** [comment]
Closing this issue assuming that it has been resolved. Feel free to reopen it if the problem appears.
